internal/providers/azure/azure.go

```diff
@@ -126,3 +144,51 @@ func isCdromPresent(logger *log.Logger, devicePath string) bool {
 
 	return (status == CDS_DISC_OK)
 }
+
+// isConfigFsValid checks if the fs is a known supported filesystem on Azure and
+// its known variants.
+func isConfigFsValid(fsType string) error {
+	switch fsType {
+	case CDS_FSTYPE_ISO9660:
+		fallthrough
+	case CDS_FSTYPE_UDF:
+		return nil
+	default:
+		return fmt.Errorf("%v is an unsupported Azure config filesystem", fsType)
+	}
+}
+
+// getConfigFsType returns the filesystem type and error for the device path.
+// Reading the db is cleaner than shelling out and it does not require
+// that system-udevd.service be running (e.g. a restart).
+// Udev entries for disks are found in /run/udev/data/b<major>:<minor>.
+func getConfigFsType(logger *log.Logger, devicePath string) (string, error) {
```

Comment:
Let's just use `GetFilesystemInfo`, since that code already exists.

Reply:
Indeed...that is the correct path.

Reply:
@bgilbert `GetFilesystemInfo` is a nicer interface too. Thanks for the tip. I'll re-test with that.